chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -836,12 +836,22 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             for signature in tx.signatures() {
                 match signature {
                     common::chain::signature::inputsig::InputWitness::NoSignature(data) => {
+                        if !self.chain_config.data_in_no_signature_witness_allowed()
+                            && data.is_some()
```

Comment:
Is it intended that this condition check data? As a result in mainnet it will be allowed to use NoSignature(None) which contradicts with the chain_config method name